go/internal/services/keys/validation.go

```diff
@@ -37,9 +42,13 @@ func (k *KeyVerifier) withCredits(ctx context.Context, cost int32) error {
 		return err
 	}
 
-	k.Key.RemainingRequests = sql.NullInt32{Int32: usage.Remaining, Valid: usage.Remaining >= 0}
+	// Handle insufficient credits: any negative value means insufficient credits
+	// In this case, we want to set RemainingRequests to 0 and mark as invalid
 	if !usage.Valid {
+		k.Key.RemainingRequests = sql.NullInt32{Int32: 0, Valid: true}
```

Comment:
that's not necessarily correct.

imagine this scenario:
1. A key has 5 credits remaining
2. A verification happens with `{credits: {cost: 10}}`

Here we should reject it, but the remaining credits are 5, not 0